packages/gatsby-source-contentful/src/plugin-options.js

```diff
@@ -0,0 +1,32 @@
+const defaultOptions = {
+  host: `cdn.contentful.com`,
+
+  // default is always master
+  environment: `master`,
+}
+
+const validateOptions = options => {
```

Comment:
Why not use joi or yup here? I've wanted to introduce something like that!

I use it here -> https://github.com/DSchau/gatsby-source-s3/blob/master/src/plugin-options.js

Reply:
I didn't want to pull extra dependencies just for checking 2 required fields but `joi` will work here as we already use it with gatsby (plus it will catch typos in fieldnames)